### ISSUE TITLE
Plateau detection should read val CE, and its bar sits below the noise floor

### ISSUE THREAD

**Issue by MatthewLacerda2** [MatthewLacerda2/TinyRefinementModel#184]
Split out of #182, which stopped a CE plateau from ending a pretraining run. This is the follow-up: making the plateau *signal* itself trustworthy.

## Two separate problems

**1. It reads train CE, which the curriculum moves underneath it.** The mixture shifts 85/10/5 -> 35/40/25 across `CURRICULUM_STEPS=10000`, so the data gets harder as the model gets better. On the #157 run, train CE drifted *up* from 3.20 to 3.36 over several hours purely from that. Comparing CE at step 5,000 against CE at step 1,000 compares two different distributions — the unmatched comparison rule 2 warns about, wired into a detector.

Val CE is the honest signal: fixed rows, fixed depth 4, fineweb-edu only, no curriculum. Comparable across time by construction.

**2. But the bar is below the noise floor.** Measured on the #157 run:

```
val CE, last 20 intervals:  mean change -0.0028   stdev 0.0111
min_delta (LossMonitor):    0.005
```

`min_delta` is less than half the per-interval noise, so on val CE the detector would be reading noise. Fixing that means a bigger probe — and `EVAL_ROWS = 4` is deliberately fixed. From `config.py`:

> Sizing or chunking the eval slice by a training throughput knob would silently redefine what "val CE" means and break comparability with every number already recorded (the champion's 4.7092, the #17 noise floor of sigma~0.03, the time machine's +/-0.06 reproduction check).

So this is a **research-doctrine tradeoff, not a bug fix**: widening the probe buys a usable plateau signal and costs comparability with every recorded val CE. Hence `plan`.

## Options

- A **second, larger probe** used only for plateau detection, leaving the reported `val_ce` untouched at 4 rows. Costs a little GPU per validation; keeps every recorded number comparable. Probably the right answer.
- Widen `EVAL_ROWS` and accept a discontinuity, recording the changeover so old and new numbers are never compared.
- Keep 4 rows and raise `min_delta` above the noise (~0.03+). Cheapest, but a detector that only fires on a 0.03-nat stall is close to no detector.

## Worth knowing before choosing

On #157 the plateau was **real** — val CE was flat at ~4.06 too. So val CE would not have prevented that stop; #182 (making the flip opt-in) is what did. This issue is about making the *signal* meaningful, not about protecting a run — that part is already handled.

Whatever lands here should also inform #180 (logging the post-accumulation gradient norm): both are cases of telemetry measuring something adjacent to the quantity decisions are made on.

**Comment by MatthewLacerda2** [comment]
## The comparability argument no longer holds, and the probe is now the weakest instrument in the loop (2026-09-15)

The reason `EVAL_ROWS = 4` stays fixed is comparability with recorded val CEs: the 4.7092, the σ≈0.03 floor from #17, the time machine's ±0.06. Every one of those numbers belongs to the **refiner** or **reasoner**, both retired (#292 tombstones them). A plain run's val CE is not comparable to them whatever the probe size, because the architecture changed. So the cost side of the trade is now zero for every run that matters.

The benefit side got bigger than "a usable plateau signal":

- **4 rows ≈ 4k scored tokens.** Per-token CE has a spread of a few nats, so the standard error of the mean on 4k tokens is roughly 0.03–0.04 nats — the same size as the seed floor. The probe cannot separate two checkpoints at the resolution the #287 LR pair (tokens-to-target-CE, val every 16 opt steps) needs. Its readout will be noise-limited by the probe, not by the model.
- **It reads fineweb-edu only,** so it is blind to the data-mix suspect on #287 by construction.
- **It is the only held-out number the trainer ever sees,** so it is what checkpoint selection (`best_val_ce/`, #222) and the milestone report read.

Proposal: take option 2 — widen `EVAL_ROWS` to 64 (about 65k tokens, standard error ~0.01 nats), record the changeover here and in the config comment, and never compare a plain run's val CE to a pre-changeover number. Cost: 64 batch-1 forwards per validation, a few seconds every 64 opt steps. Option 1 (a second probe for the detector only) keeps a comparability nobody can use and doubles the code.

Register the number before flipping it: run the new probe against the old on the same checkpoint over a few seeds of row selection, and record its own σ so #17's floor gets replaced rather than assumed.

**Comment by MatthewLacerda2** [comment]
**Decision (owner, 2026-09-15, via Claude): option 2.** Widen `EVAL_ROWS` to 64, record the changeover in the config comment and here, never compare a plain run's val CE to a pre-changeover number. Before flipping, measure the new probe's own σ (same checkpoint, a few row-selection seeds) so #17's 0.03 floor is replaced by a measured one, not assumed. This goes **before #287**, which reads the probe every 16 opt steps and would otherwise be noise-limited by it. Dropping `plan`; the issue is executable as written above.
